Marlin/src/inc/Conditionals_post.h

```diff
@@ -2166,7 +2166,7 @@
 #if PIN_EXISTS(Z4_MAX)
   #define HAS_Z4_MAX 1
 #endif
-#if BOTH(HAS_BED_PROBE, HAS_CUSTOM_PROBE_PIN) && PIN_EXISTS(Z_MIN_PROBE)
+#if ENABLED(HAS_BED_PROBE) && PIN_EXISTS(Z_MIN_PROBE)
```

Comment:
It's true that `HAS_CUSTOM_PROBE_PIN` is handled in `pins_postprocess.h`…
```cpp
#if !HAS_CUSTOM_PROBE_PIN
  #undef Z_MIN_PROBE_PIN
  #define Z_MIN_PROBE_PIN    -1
#endif
```
…but there should be no harm in leaving it here, unless there is something I'm missing…?